dashboard/app/controllers/api/v1/pd/teachercon1819_registrations_controller.rb

```diff
@@ -1,9 +1,41 @@
 class Api::V1::Pd::Teachercon1819RegistrationsController < Api::V1::Pd::FormsController
-  load_and_authorize_resource :application, class: 'Pd::Application::ApplicationBase', id_param: :applicationId
+  load_and_authorize_resource :application, class: 'Pd::Application::ApplicationBase', id_param: :applicationId, except: :create_partner
 
   def new_form
     ::Pd::Teachercon1819Registration.new(
       pd_application_id: @application.id
     )
   end
+
+  def create_partner
+    regional_partner_id = params.try(:[], :regionalPartnerId)
+    regional_partner = RegionalPartner.find(regional_partner_id)
+    unless regional_partner && current_user && current_user.regional_partners.include?(regional_partner)
```

Comment:
Note, `find` will raise an error if the id doesn't exist, so `regional_partner` here can't be nil